File: Economia-Aplicada/Econometria/Julia/macroeconometria/ar1.jl
# Modelo AR(1) por OLS: pib_t = c + phi * pib_{t-1} + erro_t
#
# Usa apenas a biblioteca padrao do Julia (sem Pkg.add necessario).
# Rode com: julia Julia/macroeconometria/ar1.jl
# (execute a partir da pasta Econometria/)

using DelimitedFiles

dados, _ = readdlm("dados/macro_series.csv", ',', header=true)
pib = Float64.(dados[:, 2])

y = pib[2:end]
y_lag = pib[1:end-1]

X = [ones(length(y_lag)) y_lag]
beta = X \ y

println("c estimado   : ", round(beta[1], digits=4))
println("phi estimado : ", round(beta[2], digits=4))

File: Economia-Aplicada/Econometria/R/macroeconometria/ar1.R
# Modelo AR(1) por OLS: pib_t = c + phi * pib_{t-1} + erro_t
#
# Rode com: Rscript R/macroeconometria/ar1.R
# (execute a partir da pasta Econometria/)

dados <- read.csv("dados/macro_series.csv")

y <- dados$pib_crescimento[-1]
y_lag <- dados$pib_crescimento[-nrow(dados)]

modelo <- lm(y ~ y_lag)
print(summary(modelo))

cat(sprintf("\nc estimado   : %.4f\n", coef(modelo)[1]))
cat(sprintf("phi estimado : %.4f\n", coef(modelo)[2]))
